Makefile

```diff
@@ -41,33 +41,47 @@ test :
 	@go test -race $(pkgs)
 
 # Docker build 
-
-ingress_binary := $(GOPATH)/bin/feed-ingress
+ingress_build_dir := build/ingress
 template := ./nginx/nginx.tmpl
-docker_repo := skycirrus/feed-ingress
 git_rev := $(shell git rev-parse --short HEAD)
-docker_tag := "$(docker_repo):$(git_rev)"
-docker_latest := "$(docker_repo):latest"
 
-clean:
+builds := ingress dns
+
+clean :
 	@echo "== cleaning"
 	rm -rf build
 
 copy : build
 	@echo "== copy docker files to build/"
-	@mkdir -p build
-	cp Dockerfile build/
-	cp $(ingress_binary) build/
-	cp $(template) build/
+	
+	@for build in $(builds) ; do \
+	  build_files=$${build}_files; \
+	  build_dir=build/$$build; \
+	  binary=$(GOPATH)/bin/feed-$$build; \
+	  docker_file=Dockerfile_$$build; \
+	  mkdir -p $$build_dir; \
+	  cp $$docker_file $${build_dir}/Dockerfile; \
+	  cp $$binary $$build_dir; \
+	done
+
+	cp ${template} ${ingress_build_dir}
```

Comment:
I could have added a var with the extra files in and had an additional for loop but that seemed over kill for one file